### PR TITLE
add a hash option to override sha1 hashing

### PR DESCRIPTION
**Pull request by hakt0r**
I added non-standard functionality in order to override SHA1 hashing - mainly for custom DHT implementations, but also useful for debugging.

### CODE REVIEW

**Comment by fisch0920** [comment]
seems reasonable to me; my only question is how this would affect other areas of webtorrent which assume sha1 throughout.  possibly related to: https://github.com/multiformats/multihash

**Comment by hakt0r** [comment]
@fisch0920 The way it's implemented you could have multiple instances of DHTs in one app, using different hash-functions without interfering with each other. The networks will be incompatible, though, as in they could not understand each others hashes. This does not implement any kind of distinction, and more work would be necessary.
@mafintosh **hash** sounds reasonable as well, my only and original argument would be that digest sounded more like a function name to me. :>

**Comment by hakt0r** [comment]
Maybe I should point out that I did not touch **DHT.prototype._generateToken** because changing the algorithm here seemed like an intrusion into the protocol with no apparent benefit.

**Comment by feross** [comment]
LGTM, but I agree with @mafintosh that the param name should be `hash`.

@fisch0920 If the `hash` option is not provided to the `DHT` constructor than SHA1 is used, so there should be no effect on WebTorrent.

File: README.md [hunk 1]
```diff
@@ -78,6 +78,7 @@ If `opts` is specified, then the default options (shown below) will be overridde
   bootstrap: [], // bootstrap servers (default: router.bittorrent.com:6881, router.utorrent.com:6881, dht.transmissionbt.com:6881)
   host: false,    // host of local peer, if specified then announces get added to local table (String, disabled by default)
   concurrency: 16 // k-rpc option to specify maximum concurrent UDP requests allowed (Number, 16 by default)
+  fingerprint:    // function used to make fingerprints (Function, sha1 by default)
```
Comment (feross) [inline README.md]: I would change this to:

```
hash: Function   // hash function to use (Function, sha1 by default)

**Comment by hakt0r** [comment]
many thanks :) :+1: @feross @mafintosh @fisch0920

**Comment by feross** [comment]
7.5.0

**Comment by talarari** [comment]
got this._hash is not a function when trying to dht.get a mutable item
reverting to 7.4.1 fixed it for me.

the problem is in this line:
https://github.com/feross/bittorrent-dht/commit/a85bfc79408b76f8f341793033a646704c9f82df#commitcomment-21687679

**Comment by feross** [comment]
@talarari Fix released as 7.5.1

**Pull request merged**